metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py

```diff
@@ -161,6 +182,72 @@ def create_view_project_map(view_fields: List[ViewField]) -> Dict[str, str]:
     return view_project_map
 
 
+def get_view_file_path(
+    lkml_fields: List[LookmlModelExploreField], view_name: str
+) -> Optional[str]:
+    """
+    Search for the view file path on field, if found then return the file path
+    """
+    logger.debug("Entered")
+
+    for field in lkml_fields:
+        if field.view == view_name:
+            # This path is relative to git clone directory
+            logger.debug(f"Found view({view_name}) file-path {field.source_file}")
+            return field.source_file
+
+    logger.debug(f"Failed to find view({view_name}) file-path")
+
+    return None
+
+
+def create_upstream_views_file_path_map(
+    view_names: Set[str], lkml_fields: List[LookmlModelExploreField]
+) -> Dict[str, Optional[str]]:
+    """
+    Create a map of view-name v/s view file path, so that later we can fetch view's file path via view-name
+    """
+    logger.debug("Entered")
+
+    upstream_views_file_path: Dict[str, Optional[str]] = {}
+
+    for view_name in view_names:
+        file_path: Optional[str] = get_view_file_path(
+            lkml_fields=lkml_fields, view_name=view_name
+        )
+
+        upstream_views_file_path[view_name] = file_path
+
+    logger.debug("Exit")
+
+    return upstream_views_file_path
+
+
+def explore_field_set_to_lkml_fields(
+    explore: LookmlModelExplore,
+) -> List[LookmlModelExploreField]:
+    """
+    explore.fields has three variables i.e. dimensions, measures, parameters of same type i.e. LookmlModelExploreField.
+    This method creating a list by adding all field instance to lkml_fields
+    """
+    lkml_fields: List[LookmlModelExploreField] = []
+
+    if explore.fields is None:
+        logger.debug(f"Explore({explore.name}) doesn't have any field")
+        return lkml_fields
+
+    def empty_list(
+        fields: Optional[Sequence[LookmlModelExploreField]],
+    ) -> List[LookmlModelExploreField]:
+        return list(fields) if fields is not None else []
+
+    lkml_fields.extend(empty_list(explore.fields.dimensions))
```

Comment:
```suggestion
    lkml_fields.extend(list(explore.fields.dimensions) or [])
```
same for below two statements. can remove nested function empty_list. It is already ensured earlier in code flow that explore.fields is not None.

Reply:
empty_list will avoid empty list creation logic three places in the function.
empty_list is handling Not None for fields which are inside explore.fields (i.e. dimensions, measures, and parameters)

Reply:
ah, okay.

Reply:
done

Reply:
in this case, I prefer the `or []` syntax over a new function

also, nesting functions like this is generally an antipattern

Reply:
It is avoiding duplicates of  `or []` logic at three statement and more readable